### PR TITLE
Applies changes needed to serve locale files as static content

### LINKED ISSUE #11041: BBB 2.2.31 - Build 1114 - systemd_start.sh[]:  'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
A last one problem

One time it is for locales/en_US.json
Another time it is for locales/fr_FR.json

```
Dec 18 10:00:37 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
Dec 18 10:00:39 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
Dec 18 10:00:58 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
Dec 18 10:02:05 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
Dec 18 10:07:26 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
Dec 18 10:11:01 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
Dec 18 10:13:21 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale fr_FR:Error: Unknown asset: locales/fr_FR.json
Dec 18 10:23:25 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale fr_FR:Error: Unknown asset: locales/fr_FR.json
Dec 18 10:36:26 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale fr_FR:Error: Unknown asset: locales/fr_FR.json
Dec 18 10:39:27 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale fr_FR:Error: Unknown asset: locales/fr_FR.json
Dec 18 10:40:06 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
Dec 18 10:47:02 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
Dec 18 10:47:44 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
Dec 18 10:47:51 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
Dec 18 10:47:55 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
Dec 18 10:48:02 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale fr_FR:Error: Unknown asset: locales/fr_FR.json
Dec 18 10:48:03 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale fr_FR:Error: Unknown asset: locales/fr_FR.json
Dec 18 10:48:15 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale fr_FR:Error: Unknown asset: locales/fr_FR.json
Dec 18 10:48:28 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
Dec 18 10:48:34 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale fr_FR:Error: Unknown asset: locales/fr_FR.json
Dec 18 10:48:40 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale fr_FR:Error: Unknown asset: locales/fr_FR.json
Dec 18 10:48:42 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale fr_FR:Error: Unknown asset: locales/fr_FR.json
Dec 18 10:48:44 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale fr_FR:Error: Unknown asset: locales/fr_FR.json
Dec 18 10:48:46 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
Dec 18 10:49:45 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
Dec 18 10:49:51 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale fr_FR:Error: Unknown asset: locales/fr_FR.json
Dec 18 10:51:53 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
Dec 18 11:04:26 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale fr_FR:Error: Unknown asset: locales/fr_FR.json
Dec 18 11:06:11 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale fr_FR:Error: Unknown asset: locales/fr_FR.json
Dec 18 11:13:21 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
Dec 18 11:15:01 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale fr_FR:Error: Unknown asset: locales/fr_FR.json
Dec 18 11:17:31 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale fr_FR:Error: Unknown asset: locales/fr_FR.json
Dec 18 11:17:57 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale fr_FR:Error: Unknown asset: locales/fr_FR.json
Dec 18 11:20:37 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
Dec 18 11:23:35 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
Dec 18 11:25:12 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale fr_FR:Error: Unknown asset: locales/fr_FR.json
Dec 18 11:29:44 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
Dec 18 11:34:30 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale fr_FR:Error: Unknown asset: locales/fr_FR.json
Dec 18 11:39:49 bbb-host systemd_start.sh[6841]: #033[33mwarn#033[39m: 'Could not process locale en_US:Error: Unknown asset: locales/en_US.json
```

```
bbb-host log$ls -al /var/www/bigbluebutton/client/locale/ |grep fr
-rw-r--r--  1 root root 197183 May 13  2020 fr_CA_resources.swf
-rw-r--r--  1 root root 137810 May 13  2020 fr_CA_resources.swf.cache
-rw-r--r--  1 root root 198114 May 13  2020 fr_FR_resources.swf
-rw-r--r--  1 root root 145864 May 13  2020 fr_FR_resources.swf.cache
-rw-r--r--  1 root root 195407 May 13  2020 fr_resources.swf
-rw-r--r--  1 root root 131422 May 13  2020 fr_resources.swf.cache
```

```
bbb-host log$ls -al /var/www/bigbluebutton/client/locale/ |grep en
-rw-r--r--  1 root root 195454 May 13  2020 en_AU_resources.swf
-rw-r--r--  1 root root 131545 May 13  2020 en_AU_resources.swf.cache
-rw-r--r--  1 root root 195449 May 13  2020 en_resources.swf
-rw-r--r--  1 root root 131480 May 13  2020 en_resources.swf.cache
-rw-r--r--  1 root root 195350 May 13  2020 en_US_resources.swf
-rw-r--r--  1 root root 138545 May 13  2020 en_US_resources.swf.cache
```

```
$cat /etc/default/locale

#  File generated by update-locale
LANG=en_US.UTF-8
```

### PR DESCRIPTION
**Pull request by ramonlsouza**
### What does this PR do?

Applies changes needed to serve locale files as static content.

`bbb-html5.nginx` file content should also be changed to the following:

```
location /html5client/locales {
  alias /usr/share/meteor/bundle/programs/web.browser/app/locales;
}

location /html5client/compatibility {
  alias /usr/share/meteor/bundle/programs/web.browser/app/compatibility;
}

location /html5client/resources {
  alias /usr/share/meteor/bundle/programs/web.browser/app/resources;
}

location /html5client/svgs {
  alias /usr/share/meteor/bundle/programs/web.browser/app/svgs;
}

location /html5client/fonts {
  alias /usr/share/meteor/bundle/programs/web.browser/app/fonts;
}

location /html5client {
  # proxy_pass http://127.0.0.1:4100; # use for development
  proxy_pass http://poolhtml5servers; # use for production
  proxy_http_version 1.1;
  proxy_set_header Upgrade $http_upgrade;
  proxy_set_header Connection "Upgrade";
}
```

### Closes Issue(s)

closes #11041

### CODE REVIEW

**Comment by kepstin** [comment]
FYI: The top level rewrite in the nginx configuration:
```
rewrite ^/html5client/[0-9]+/(fonts|locales|compatibility|resources|svgs)/(.*) /html5client/$1/$2;
```

is applied prior to any location block matching. That rewrite removes the numeric (instance id) prefix in the patch for paths matching `fonts|locales|compatibility|resources|svgs`

As a result, in the later location block:
```
location ~ ^/html5client/(?<instanceId>[0-9]+)/(?!(fonts|locales|compatibility|resources|svgs)) {
```

The regex to exclude those paths (`(?!(fonts|locales|compatibility|resources|svgs)`) is not necessary.

File: bigbluebutton-html5/imports/startup/server/index.js [hunk 1]
```diff
@@ -12,7 +13,16 @@ import Redis from './redis';
...
 let guestWaitHtml = '';
-const AVAILABLE_LOCALES = fs.readdirSync('assets/app/locales');
+
+const meteorRoot = fs.realpathSync(`${process.cwd()}/../`);
+let applicationRoot = fs.realpathSync(`${meteorRoot}/../`);
+
+// if running on dev mode
+if (path.basename(fs.realpathSync(`${meteorRoot}/../../../`)) === '.meteor') {
+  applicationRoot = fs.realpathSync(`${meteorRoot}'/../../../../`);
+}
```
Comment (pedrobmarin) [inline bigbluebutton-html5/imports/startup/server/index.js:23]: Is there a way to avoid this dot slashes?

**Comment by basisbit** [comment]
You mention `Includes improvements on nginx configuration files, as described on the following gist` in your PR description but the diff does not contain any of these - did you forget to push some changes?

File: bigbluebutton-html5/imports/startup/server/index.js [hunk 2]
```diff
@@ -12,7 +13,16 @@ import Redis from './redis';
 import setMinBrowserVersions from './minBrowserVersion';
 
 let guestWaitHtml = '';
-const AVAILABLE_LOCALES = fs.readdirSync('assets/app/locales');
+
+const meteorRoot = fs.realpathSync(`${process.cwd()}/../`);
+let applicationRoot = fs.realpathSync(`${meteorRoot}/../`);
+
+// if running on dev mode
+if (path.basename(fs.realpathSync(`${meteorRoot}/../../../`)) === '.meteor') {
```
Comment (basisbit) [inline bigbluebutton-html5/imports/startup/server/index.js:21]: please remove this line and instead change applicationRoot to a const again and check `process.env.NODE_ENV || 'development'` for deciding on which of the two paths to assign to the constant.

**Review by basisbit — DISMISSED** [review]

**Comment by ramonlsouza** [comment]
> You mention `Includes improvements on nginx configuration files, as described on the following gist` in your PR description but the diff does not contain any of these - did you forget to push some changes?

@basisbit the nginx configuration files are not in this repository, I just linked them because they are related to the PR, so @antobinary could see the changes and apply them to the build if needed

**Pull request merged**